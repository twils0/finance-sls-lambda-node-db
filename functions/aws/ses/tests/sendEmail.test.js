const AWS = require('aws-sdk'); // eslint-disable-line

const sendEmailExport = require('../sendEmail');

const sendEmailModule = sendEmailExport.sendEmail;

const sendEmail = jest.fn();

global.console.log = jest.fn();

AWS.SES = jest.fn(() => ({
  sendEmail,
}));

const fromEmail = 'support@dynamictriggers.com';
const fromDomain = 'test.com';
const subject = 'testSubject';
const body = 'testBody';
const emailAddresses = ['toTest0@test.com', 'toTest1@test.com'];

describe('functions', () => {
  describe('aws', () => {
    describe('ses', () => {
      describe('sendEmail', () => {
        afterEach(() => {
          global.console.log.mockReset();
          sendEmail.mockReset();
        });

        it('fails and returns Email Not Sent, no body', async () => {
          const email = {
            fromEmail,
            fromDomain,
            subject,
          };
          sendEmail.mockImplementation((params, callback) => callback(null, null));

          const result = await sendEmailModule(email, emailAddresses);

          expect(result).toEqual(null);
          expect(global.console.log).toMatchSnapshot();
        });

        it('fails and returns Email Not Sent, invalid fromEmail', async () => {
          const email = {
            fromEmail: 'test@test.com',
            fromDomain,
            subject,
            body,
          };
          sendEmail.mockImplementation((params, callback) => callback(null, null));

          const result = await sendEmailModule(email, emailAddresses);

          expect(result).toEqual(null);
          expect(global.console.log).toMatchSnapshot();
        });

        it('fails and returns Email Not Sent, no emailAddresses', async () => {
          const email = {
            fromEmail,
            fromDomain,
            subject,
            body,
          };

          sendEmail.mockImplementation((params, callback) => callback(null, null));

          const result = await sendEmailModule(email, null);

          expect(result).toEqual(null);
          expect(global.console.log).toMatchSnapshot();
        });

        it('sendEmail fails and returns unexpected error', async () => {
          const email = {
            fromEmail,
            fromDomain,
            subject,
            body,
          };
          const error = {
            code: 'testCode',
            message: 'testMessage',
          };

          sendEmail.mockImplementation((params, callback) => callback(error, null));

          const result = await sendEmailModule(email, emailAddresses);

          expect(result).toEqual(null);
          expect(global.console.log).toMatchSnapshot();
        });

        it('correctly sends emails', async () => {
          const email = {
            fromEmail,
            fromDomain,
            subject,
            body,
          };

          sendEmail.mockImplementation((params, callback) => callback(null, null));

          const result = await sendEmailModule(email, emailAddresses);

          expect(result).toEqual(null);
          expect(global.console.log).not.toBeCalled();
        });
      });
    });
  });
});
