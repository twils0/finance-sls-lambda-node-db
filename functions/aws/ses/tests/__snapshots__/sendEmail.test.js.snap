// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`functions aws ses sendEmail fails and returns Email Not Sent, invalid fromEmail 1`] = `[MockFunction]`;

exports[`functions aws ses sendEmail fails and returns Email Not Sent, no body 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "Email Not Sent - no body provided",
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`functions aws ses sendEmail fails and returns Email Not Sent, no emailAddresses 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "Email Not Sent - no email addresses provided",
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`functions aws ses sendEmail sendEmail fails and returns unexpected error 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "unexpected error - failed to send email - toTest0@test.com",
      "
",
      Object {
        "code": "testCode",
        "message": "testMessage",
      },
    ],
    Array [
      "unexpected error - failed to send email - toTest1@test.com",
      "
",
      Object {
        "code": "testCode",
        "message": "testMessage",
      },
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;
